app/core/auth/__init__.py

```diff
@@ -76,21 +76,18 @@ async def wrapper(*args, **kwargs):
         supabase: Client = create_client(supabase_url, supabase_key)
         try:
             user_data = supabase.auth.get_user(access_token_str)
-        except gotrue.errors.AuthApiError as e:
-            raise PermissionException("Unauthorized access. Invalid token.") from e
-        result = supabase.table('userTypes').select('''
-                sources
-                '''
-            ).eq(
-            'user_type', user_data.user.user_metadata.get('user_type')
-            ).limit(1).single().execute()
-        labels = result.data.get('sources')
-
-        if not labels:
-            raise PermissionException(
-            "Unauthorized access. User does not have access to any sources." 
-            # Or we could just give open access sources?
-            )
+
+        except gotrue.errors.AuthApiError as e: # The user is not logged in
+            labels = ['public']
+
+        else:
+            result = supabase.table('userTypes').select('''
+                    sources
+                    '''
+                ).eq(
+                'user_type', user_data.user.user_metadata.get('user_type')
+                ).limit(1).single().execute()
+            labels = result.data.get('sources')
 
         kwargs['label'] = labels
         # Proceed with the original function call and pass the sources to it
```

Comment:
For this "public" label to work, all different kinds sources we expect to open up will need to have this one label "public".
How I had in my mind is some thing this,
You upload a bible, with label "ESV-Bible", a commentry with label, "Some-open-commentary" and such different types of resources with different labels.
Now in the access mapping table where we list all users and the labels they can access
1. either add these open labels to all users
2. or have a special row for "all_users" or "no-logged-in-user" etc and list these public labels there. 
I think such a approach will allow us to maintain the collections identity by unique labels and also move around things from public and propriteory if needed easily, just by changing the user-label table.